subsys/bluetooth/common/Kconfig

```diff
@@ -184,11 +184,34 @@ config BT_BUF_CMD_TX_SIZE
 
 config BT_BUF_CMD_TX_COUNT
 	int "Number of HCI command buffers"
-	default 2
-	range 2 64
+	default 2 if BT_HCI_ACL_FLOW_CONTROL
+	default 1
+	range 1 64
 	help
 	  Number of buffers available for outgoing HCI commands from the Host.
 
+	  HCI Controllers may not support Num_HCI_Command_Packets > 1, hence
+	  default to 1 when not enabling Controller to Host data flow control,
+	  BT_HCI_ACL_FLOW_CONTROL.
```

Comment:
This paragraph isn't completely clear to me. Even if a controller supports ncmd > 1, the Zephyr host has a command queue and will always round down the value to 1, or rather, treat it as a boolean:
https://github.com/zephyrproject-rtos/zephyr/blob/ddd795a028b1a4f9e9fa672e7791b1750f34d690/subsys/bluetooth/host/hci_core.c#L2499-L2503

The other help text paragraphs sort of make sense, but I just wanted to make sure the above host implementation detail is clear, so that this PR isn't making any false assumptions (since I haven't quite wrapped my head around all the details of it).

Reply:
Yes, Zephyr Host only implements `ncmd` = 1. See,
https://github.com/zephyrproject-rtos/zephyr/blob/ddd795a028b1a4f9e9fa672e7791b1750f34d690/subsys/bluetooth/host/hci_core.c#L4340-L4348

I guess it is not difficult to use the same semaphore and we able to support `ncmd` > 1, but we do not have support in native Controller today for that though.

Reply:
> the Zephyr host has a command queue and will always round down the value to 1, or rather, treat it as a boolean:

@jhedberg I believe the implementation "can"  support `ncmd` > 1, but today `ncmd` in the `evt` is only used as a bool to pause the queuing, but the `ncmd_sem` if initialized correctly would support `ncmd` > 1. But we are not going towards that in this PR.

Reply:
> @jhedberg I believe the implementation "can" support `ncmd` > 1, but today `ncmd` in the `evt` is only used as a bool to pause the queuing, but the `ncmd_sem` if initialized correctly would support `ncmd` > 1. But we are not going towards that in this PR.

That's not quite true, because of `bt_dev.sent_cmd`. That would need to be transformed into a list or array, which in turn would introduce some additional processing for command complete packets (right now this is O(1) when checking of the completed opcode and buffer match the last sent command (see `hci_cmd_done()` in `hci_core.c`).

Reply:
Agree. I can update the comment if required to state that even if a Controller would support `ncmd` > 1, the Host does not have the required implementation for do `ncmd` > 1.